lib/kamal/commands/docker.rb

```diff
@@ -19,6 +19,12 @@ def superuser?
     [ '[ "${EUID:-$(id -u)}" -eq 0 ] || command -v sudo >/dev/null || command -v su >/dev/null' ]
   end
 
+  # If we're not root and not already in the docker group
+  # add us to the docker group and terminate all our current sessions
+  def add_group
+    [ '[ "${EUID:-$(id -u)}" -eq 0 ] || id -nG "${USER:-$(id -un)}" | grep -qw docker || { sudo usermod -aG docker "${USER:-$(id -un)}" && kill -HUP ${PPID:-ps -o ppid= -p $$}; }' ]
```

Comment:
Also, `/etc/ssh/sshd_config` should set `PermitRootLogin no` and `PasswordAuthentication no`. What you think?

Reply:
`newgrp` won't work. The persistent session is on the parent user `sshd` process, which will remain with the original group set, in a different terminal session. You have to terminate that to force a re-authentication to pick up the new group set. 

Even `exec newgrp` won't do the trick. There's no persistent shell process to replace

Reply:
> Also, /etc/ssh/sshd_config should set PermitRootLogin no and PasswordAuthentication no. What you think?

I don't think we want to get too far into configuring the servers. The docker group change makes sense here because it is only used when we have just installed docker and we know it's going to be a problem.

> newgrp won't work. The persistent session is on the parent user sshd process, which will remain with the original group set, in a different terminal session. You have to terminate that to force a re-authentication to pick up the new group set.

Ok I think signalling is probably ok then - it's a bit of a shame it has to done that way, but I think it's worth it for making this a smoother process.

Reply:
So I think we just need to add `-n` to sudo, and then make sure there's a nice error message if the user doesn't have sudo access.